src/win.c

```diff
@@ -1117,6 +1139,27 @@ void win_update_wintype(session_t *ps, struct managed_win *w) {
 	}
 }
 
+/**
+ * Update window wm_state.
+ */
+void win_update_wm_state(session_t *ps, struct managed_win *w) {
+	const win_wmstate_t wmstate_old = w->wm_state;
+
+	// Detect window wm_state here
+	w->wm_state = wid_get_prop_wm_state(ps, w->client_win);
+
+	if (w->wm_state != wmstate_old) {
+		win_on_factor_change(ps, w);
```

Comment:
Should set the FACTOR_CHANGE flag instead.

Reply:
Right, missed that one. Same would apply for `win_update_wintype()` above.

Reply:
While we're at it: I think [`win_update_opacity_target`](https://github.com/yshui/picom/blob/fb38bf021e1928b62bb495f06899b89be430b98c/src/win.c#L369) can be replaced as well.